src/include/sof/audio/component.h

```diff
@@ -342,8 +342,12 @@ void sys_comp_host_init(void);
 void sys_comp_mixer_init(void);
 void sys_comp_mux_init(void);
 void sys_comp_switch_init(void);
+#ifdef CONFIG_COMP_VOLUME
```

Comment:
Not sure if makes sense to maintain the central list here makes sense if it is more dynamic at the compile time. Keeping the declaration of individual comp init entry in its header file included by the central component initialization code (pre-processed with the config option) should be enough I guess.

Reply:
I have removed ifdefs from component.h as we discussed today and I will make additional PR/commit for this init functions